src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -36,6 +36,9 @@ internal static class ThreadPoolGlobals
         public static volatile bool vmTpInitialized;
         public static bool enableWorkerTracking;
 
+        public const int wsqNowActive = 2; // Require at least 2 full observations of empty wsq to transition to inactive
+        public static int wsqActive;
```

Comment:
I think wsqActive probably needs to be an member variable of WSQL, in case there is more than one thread pool.  One per appdomain for instance.